File: Java/src/com/leetcode/Easy/LongestPalindrome.java
package com.leetcode.Easy;

import java.util.*;

// 409. Longest Palindrome
public class LongestPalindrome {
    public int longestPalindrome(String s) {
        if (s == null || s.length() == 0)
            return 0;

        int count = 0;
        Set<Character> set = new HashSet<>();
        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            if (set.contains(ch)) {
                set.remove(ch);
                count++;
            } else {
                set.add(ch);
            }
        }

        return set.isEmpty() ? count * 2 : count * 2 + 1;
    }
}

File: JavaScript/Easy/LongestPalindrome.js
/**
 * 409. Longest Palindrome
 * @param {string} s
 * @return {number}
 */
const longestPalindrome = function (s) {
  if (!s || s.length === 0) {
    return 0;
  }

  let count = 0;
  let set = new Set();

  for (let i = 0; i < s.length; i++) {
    let ch = s.charAt(i);
    if (set.has(ch)) {
      set.delete(ch);
      count++;
    } else {
      set.add(ch);
    }
  }

  return set.size === 0 ? count * 2 : count * 2 + 1;
};